### DISCUSSION TITLE
Docker performance

### DISCUSSION THREAD

**Discussion by orzechszek** [theengs/gateway#102] [General]
Hi,

Does anyone noticed lower performance with docker instance? 

With docker conf:
{
    "host": "192.168.xx.xxx",
    "pass": "",
    "user": "",
    "port": 1883,
    "publish_topic": "home/TheengsGateway/BTtoMQTT",
    "subscribe_topic": "home/+/BTtoMQTT/undecoded",
    "publish_all": true,
    "ble_scan_time": 5,
    "ble_time_between_scans": 3,
    "log_level": "DEBUG",
    "discovery": true,
    "hass_discovery": true,
    "discovery_topic": "homeassistant/sensor",
    "discovery_device_name": "TheengsGateway",
    "discovery_filter": "[IBEACON,GAEN,MS-CDP]",
    "scanning_mode": "active",
    "adapter": "hci0"
}

and similar installation for python:
{
    "adapter": "",
    "ble_scan_time": 5,
    "ble_time_between_scans": 3,
    "discovery": 1,
    "discovery_device_name": "TheengsGateway",
    "discovery_filter": [
        "IBEACON",
        "GAEN",
        "MS-CDP"
    ],
    "discovery_topic": "homeassistant/sensor",
    "hass_discovery": 1,
    "host": "192.168.xx.xxx",
    "log_level": "DEBUG",
    "pass": "",
    "port": 1883,
    "publish_all": 1,
    "publish_topic": "home/TheengsGateway/BTtoMQTT",
    "scanning_mode": "active",
    "subscribe_topic": "home/TheengsGateway/+",
    "time_format": 0,
    "time_sync": [],
    "user": ""
}

I'm getting more data from my beacons. With docker instance I'm loosing quite often an iteration of a scan (5 seconds). WIth python I see almost everything.

I found another question - anybody verified if all parameters are interpreted correctly ? Maybe it's a problem with publish_all ? 

Had anyone such behavior?

**Comment by koenvervloesem** [comment]
I haven't looked yet into the behaviour on Docker (and I guess the Home Assistant addon would have the same problem then), but I don't think we can do anything about it. Theengs Gateway just uses the underlying Bluetooth stack exposed by the host OS. What's your Docker command or Docker Compose configuration to launch Theengs Gateway?

The `publish_all` is handled incorrectly indeed. This has been fixed in https://github.com/theengs/gateway/pull/96 so this will be in the next release.
